src/lib.rs

```diff
@@ -678,11 +741,26 @@ fn has_inline_attribute(attrs: &[&syn::Attribute]) -> bool {
     })
 }
 
+struct MatchVisitor {
+    name: Ident,
+    args: Vec<syn::Expr>,
+}
+
+impl VisitMut for MatchVisitor {
+    fn visit_arm_mut(&mut self, arm: &mut syn::Arm) {
+        let body = arm.body.clone();
+        let name = &self.name;
+        let args = &self.args;
+        arm.body = syn::parse2(quote::quote! { #body.#name(#(#args),*) }).unwrap();
```

Comment:
Just in case there's something bad here in the parsed code, could you instead return `Result` from the method, and if this transformation fails, just use the normal delegation? (`match <expr> { }.delegate()`) And maybe print a warning.

Reply:
got it